automation-packages/load-testing-okhttp/src/main/java/step/examples/loadtesting/okhttp/OkHttpKeywords.java

```diff
@@ -1,27 +1,170 @@
 package step.examples.loadtesting.okhttp;
 
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.Response;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.*;
+import step.functions.io.AbstractSession;
 import step.handlers.javahandler.AbstractKeyword;
 import step.handlers.javahandler.Keyword;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public class OkHttpKeywords extends AbstractKeyword {
+    private static final boolean WORKAROUND = true;
 
-    @Keyword(name = "OpenCart home")
-    public void openCartHome() throws InterruptedException {
-        OkHttpClient client = new OkHttpClient();
+    private static final String OKHTTP_CLIENT = "okhttpClient"; // not working, keeps running into the following problems regardless of session etc.:
+    // errorSummary=class okhttp3.OkHttpClient cannot be cast to class okhttp3.OkHttpClient (okhttp3.OkHttpClient is in unnamed module of loader step.grid.contextbuilder.JavaLibrariesClassLoader @514c7ee5; okhttp3.OkHttpClient is in unnamed module of loader step.grid.contextbuilder.JavaLibrariesClassLoader @69636681)
+    private static final String OKHTTP_COOKIES_WORKAROUND = "okhttpClient_Cookies_Workaround";
 
-        Request request = new Request.Builder()
-                .url("https://opencart-prf.exense.ch/")
-                .build();
+    @Keyword(name = "okhttp Init")
+    public void okhttpClientInit() {
```

Comment:
I would perform the initialization of the client implicitly in the Keywords in order not to expose a technical KW in the plans

Reply:
Done